app/controllers/tag_controller.rb

```diff
@@ -8,7 +8,7 @@ def index
 
     @title = I18n.t('tag_controller.tags')
     @paginated = true
-    @order_type = params[:order] == "desc" ? "asc" : "desc"
+    @order_type = params[:order].blank? || params[:order] == "desc" ? "asc" : "desc"
```

Comment:
Ooh, wow the original code is quite hard to interpet, huh?

`@order_type = params[:order] == "desc" ? "asc" : "desc"`

I'm actually not sure the order of the operations... it almost looks like it would assign the value `true` or `false` (the output of the `==` evaluation) to `@order_type`? But i must be confused. 

What if we broke this up onto different lines, or added some parentheses to make it more readable? 

What i did to try to understand was to run the following in my `irb` console:

```ruby
Somerville:~ warren$ irb
irb(main):001:0> "desc" == "desc" ? "asc" : "desc"
=> "asc"
irb(main):002:0> "asc" == "desc" ? "asc" : "desc"
=> "desc"
irb(main):003:0> "desc".blank? || "desc" == "desc" ? "asc" : "desc" 
NoMethodError: undefined method `blank?' for "desc":String
	from (irb):3
	from /usr/bin/irb:11:in `<main>'
irb(main):004:0> "desc".empty? || "desc" == "desc" ? "asc" : "desc"
=> "asc"
irb(main):008:0> "asc".empty? || "asc" == "desc" ? "asc" : "desc"
=> "desc"
irb(main):007:0> "".empty? || "" == "desc" ? "asc" : "desc"
=> "asc"
```

So one thing is that I think it would have to be `.empty?` instead of `.blank?`? (i believe, although maybe the Ruby version affects this?)

But, is the purpose here to have it default to "desc" if it's empty? If so maybe something simpler like this could work? 

```ruby
param = "desc" if param.empty?
```

Reply:
Ohh hi @jywarren , thanks for taking out the time to review this. 
> Ooh, wow the original code is quite hard to interpet, huh?
Haha, true. After all the setting up and debugging, it all came down to this line, in the end. If only one could see my face, then.

> I'm actually not sure the order of the operations... it almost looks like it would assign the value true or false (the output of the == evaluation) to @order_type? But i must be confused.

From what I observed in the code is order_type is to toggle between the current default ordering of the tags. Hence it's "asc" when the current value(that is passed in `params[:order]`) has a value "desc" and vice versa.

> What if we broke this up onto different lines, or added some parentheses to make it more readable?

rubocop (the linter for ruby used here), does not recommend using parenthesis for ternary operators, but can surely change it if you want. let me know in response.

> "desc".blank? || "desc" == "desc" ? "asc" : "desc" 
> So one thing is that I think it would have to be .empty? instead of .blank?? (i believe, although maybe the Ruby version affects this?)
 This is because `.blank?` is a rails method and not a `ruby` method. I prefer it to be `blank?`, mainly because `"".blank?` is true and `"   ".blank?` is also true. Hence every string with only whitespaces is considered to be `blank`. Just a convention, I personally follow and prefer. But again, `empty?` also works, just that `"   ".empty?` is false, hence should not be preferred, IMO.

Thoughts?

Reply:
@jywarren ?